pkg/auth/manager.go

```diff
@@ -5,27 +5,56 @@ package auth
 
 import (
 	"fmt"
-	"strconv"
+	"net"
+	"time"
 
-	"github.com/cilium/cilium/pkg/endpointmanager"
-	"github.com/cilium/cilium/pkg/logging/logfields"
-	"github.com/cilium/cilium/pkg/maps/ctmap"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ip"
 	"github.com/cilium/cilium/pkg/monitor"
 	"github.com/cilium/cilium/pkg/policy"
-	"github.com/cilium/cilium/pkg/u8proto"
 )
 
 type authManager struct {
-	endpointManager endpointmanager.EndpointsLookup
-	authHandlers    map[policy.AuthType]authHandler
+	ipCache               ipCache
+	authHandlers          map[policy.AuthType]authHandler
+	datapathAuthenticator datapathAuthenticator
 }
 
+// ipCache is the set of interactions the auth manager performs with the IPCache
+type ipCache interface {
+	GetHostIP(ip string) net.IP
+}
+
+// authHandler is responsible to handle authentication for a specific auth type
 type authHandler interface {
-	authenticate() error
+	authenticate(*authRequest) (*authResponse, error)
 	authType() policy.AuthType
 }
 
-func newAuthManager(epMgr endpointmanager.EndpointsLookup, authHandlers []authHandler) (*authManager, error) {
+type authRequest struct {
+	localIdentity  identity.NumericIdentity
+	remoteIdentity identity.NumericIdentity
+	remoteHostIP   net.IP
+}
+
+type authResponse struct {
```

Comment:
What is the difference between the `authResponse` and the `authResult`?

Reply:
(I seem to have found it but would be nice to clarify in the code for new readers)

Reply:
`authResult` is more or less the sum of all information at the end of the authentication which gets written back to a BPF map. in the current case (using `conntrack` map) this is more related to connection information. when changing this towards `auth` map in the future we need to pass all these relevant information down.

`authRequest` & `authResponse` -> belong to `authHandler`
`authResult` -> belongs to `datapathAuthenticator`

Reply:
documented on the struct and moved connectioninfo and dropnotify pointers into the struct.

this might needs some cleanup after we replaced the ct map with the auth map - as they need totally different information. we should be able to delete connectioninfo & dropnotify in the near future.